backend/src/datasources/posts.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { createTestClient } from 'apollo-server-testing'
+import { gql } from 'apollo-server'
+import server from '../server'
+import { posts, users} from '../seed-data'
+import PostDatasource from './post.datasource.js'
+
+const { query, mutate } = createTestClient(server)
+
+function toRawPost(post){
+    return{
+        id : post.id, 
+        title: post.title, 
+        userName: post.author.name,
+        voters: []
+    }
+}
+
+describe('queries', () =>{
+    describe('POSTS', () => {
+        const POSTS = gql`
+            query {
+                posts {
+                    id
+                    title
+                    author {
+                        name
+                    }
+                    
+                }
+            }` 
+    
+        
+        it('returns full array', async () =>{
+            const response = query({ query: POSTS}).then(res => res.data.posts.map(toRawPost));
+            await expect(response).resolves.toMatchObject(posts)
```

Comment:
It's not obvious what the expected values are. Depending on what you're interested in you could do:
```suggestion
            // for values
            await expect(response).resolves.toEqual([
              expect.objectContaining({
                // minimum of expected attributes go here
              })
            ])

            // for number of returned items
            const { data: { posts } } = await query({ query: POSTS})
            expect(posts).toHaveLength(42)
```